feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go

```diff
@@ -264,67 +247,73 @@ func testIPv4BackUpSwitch(ctx context.Context, t *testing.T, args *testArgs) {
 	// validate programming using AFT
 	aftCheck(t, args.dut, dstPfx, []string{"192.0.2.6", "192.0.2.10"})
 	// Validate traffic over primary path port2, port3
-	validateTrafficFlows(t, args.ate, BaseFlow, false, []string{"port2", "port3"})
+	validateTrafficFlows(t, args.ate, args.top, BaseFlow, false)
 
 	//shutdown port2
-	flapinterface(t, args.ate, "port2", false)
-	defer flapinterface(t, args.ate, "port2", true)
+	flapinterface(t, args.dut, "port2", false)
+	defer flapinterface(t, args.dut, "port2", true)
 	// validate programming using AFT
 	aftCheck(t, args.dut, dstPfx, []string{"192.0.2.10"})
 	// Validate traffic over primary path port3
-	validateTrafficFlows(t, args.ate, BaseFlow, false, []string{"port3"})
+	validateTrafficFlows(t, args.ate, args.top, BaseFlow, false)
 
 	//shutdown port3
-	flapinterface(t, args.ate, "port3", false)
-	defer flapinterface(t, args.ate, "port3", true)
+	flapinterface(t, args.dut, "port3", false)
+	defer flapinterface(t, args.dut, "port3", true)
 	// validate programming using AFT
 	aftCheck(t, args.dut, dstPfx, []string{"192.0.2.14"})
 	// validate traffic over backup
-	validateTrafficFlows(t, args.ate, BaseFlow, false, []string{"port4"})
+	validateTrafficFlows(t, args.ate, args.top, BaseFlow, false)
 }
 
 // createFlow returns a flow from atePort1 to the dstPfx
-func createFlow(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology, name string) *ondatra.Flow {
-	srcEndPoint := top.Interfaces()[atePort1.Name]
-	dstEndPoint := []ondatra.Endpoint{}
-	for intf, intf_data := range top.Interfaces() {
-		if intf != "atePort1" {
-			dstEndPoint = append(dstEndPoint, intf_data)
-		}
-	}
-	hdr := ondatra.NewIPv4Header()
-	hdr.WithSrcAddress(dutPort1.IPv4).DstAddressRange().WithMin(dstPfxMin).WithCount(1)
-
-	flow := ate.Traffic().NewFlow(name).
-		WithSrcEndpoints(srcEndPoint).
-		WithDstEndpoints(dstEndPoint...).
-		WithHeaders(ondatra.NewEthernetHeader(), hdr).WithFrameSize(300)
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config, name string) string {
+
+	flow := top.Flows().Add().SetName(name)
+	flow.Metrics().SetEnable(true)
+	e1 := flow.Packet().Add().Ethernet()
+	e1.Src().SetValue(atePort1.MAC)
+	flow.TxRx().Port().SetTxName(atePort1.Name)
```

Comment:
What the behavior if there is no Rx name set here?

Reply:
In this case flow statistics could not be retrieved on the OTG side but the traffic will be generated. We configure port 1 as Tx and the packets may be received by one or more Rx Ports. At this time OTG cannot accept more than 1 destination as Rx.